parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -337,14 +349,22 @@ pub mod pallet {
 			Ok(ticket)
 		}
 
-		fn submit(ticket: Self::Ticket) -> Result<MessageHash, SubmitError> {
-			// Make sure the bridge not halted
-			Self::ensure_not_halted().map_err(|_| SubmitError::BridgeHalted)?;
-			T::MessageQueue::enqueue_message(
-				ticket.message.as_bounded_slice(),
-				AggregateMessageOrigin::Parachain(ticket.origin),
-			);
-			Self::deposit_event(Event::MessageQueued { id: ticket.id });
+		fn submit(ticket: Self::Ticket, priority: Priority) -> Result<MessageHash, SubmitError> {
+			match priority {
+				Priority::Normal => {
+					Self::ensure_not_halted().map_err(|_| SubmitError::BridgeHalted)?;
+					T::MessageQueue::enqueue_message(
+						ticket.message.as_bounded_slice(),
+						AggregateMessageOrigin::Parachain(ticket.origin),
+					);
+					Self::deposit_event(Event::MessageQueued { id: ticket.id });
+				},
+				Priority::High => {
+					Self::do_process_message(&ticket.message.as_bounded_slice(), priority)
```

Comment:
This call will fail if `MessageLeaves` is already filled up by the time the code reaches here. As long as the number of queued messages is greater than the maximum length of `MessageLeaves`, this edge case will always be hit.

Lets trace the execution of a block:
1. `on_initialize` is called to reset storage
2. `MessageQueue::on_initialize` delivers messages to the message processor `process_message`
3. `MessageLeaves` gets filled up with low priority messages
4. Governance submits an upgrade message. This fails as there is no more space in `MessageLeaves`.

Reply:
Thanks for pointing it out. Yes, I also realize it and revamp in https://github.com/Snowfork/snowbridge/commit/42692ab2fd457618d325fb7ce87e89f08bd42c8f

This only concern is about how to get the priority, for me seems not relevant to the parachain(I would assume it's still possible that low priority message from bridge hub or high priority from sibling chains), it's more like an [extended attribute](https://github.com/Snowfork/snowbridge/pull/949/files#r1357042670) of the command.